packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -265,21 +261,26 @@ class DataSourceWithBackend<
   }
 
   /**
-   * Apply template variables for explore
+   * Filters out queries that are empty or hidden. Used when running queries through backend.
+   * @returns `true` if the query is not hidden and its expression is not empty; `false` otherwise.
    */
-  interpolateVariablesInQueries(queries: TQuery[], scopedVars: ScopedVars, filters?: AdHocVariableFilter[]): TQuery[] {
-    return queries.map((q) => this.applyTemplateVariables(q, scopedVars, filters) as TQuery);
+  filterQuery(query: TQuery): boolean {
+    if (query.hide) {
+      return false;
```

Comment:
It's a bit weird that Grafana doesn't filter out hidden queries by default - we should probably have done it a long time ago, so I'm not against this. However, technically this is a breaking change as it would start filtering out queries that were previously not hidden because the data source had not implement the `filterQuery` method. 

Again, I'm not against this but if we're doing this change we need to be very loud about it, provide proper migration guidelines and also do it in a separate PR.